.changeset/olive-cups-kiss.md

```diff
@@ -0,0 +1,34 @@
+---
+'sku': major
+---
+
+Change the CLI to commander.js.
+
+**BREAKING CHANGE**
+
+SKU now uses [commander.js](https://github.com/tj/commander.js) for its CLI. The CLI options and commands are now more narrowly defined and scoped. Some combinations of options that were previously unintended may no longer work.
+
+The new `-h, --help` flag now shows the help output generated by commander.js and shows the available commands and options.
+
+The following option flags are now available in their respective scope
+
+#### global options:
+
+- the `-e, --environment` option now only accepts a valid node.js environment.
+- the `-c, --config option` accepts the path to the sku.config.js file.
+- the `-p, --packageManager` option accepts a string for the packagemanger you're using.
+- the `-d, --debug` option sets debugging to true.
+- __new:__ the `-h, --help` flag now shows you the full CLI usage of any command or subcommand
+- __new:__ the `-v, --version` flag now shows you the version of the CLI
+
+#### scoped options:
+
+- the `build`, `build-ssr`, `start`, and `start-ssr commands` have the following options
+  - `-s, --stats` the webpack preset to use. This option now only accepts a valid webpack stats preset.
+- the `serve` command now has the following options
+  - `--port` the port you want to serve on. This option accepts a port as a number.
+  - `--site` the url you want to serve on. This option accepts a site as a string.
+- the `test`, and `translation compile` commands have the following options
+  - `-w, --watch` makes the commands watch for changes.
```

Comment:
I think `--watch` should really only be mentioned for `translations compile`. As far as `sku test` is concerned it's just forwarding arguments to `jest`.

I think the only reason it's explicitly forwarded to `jest` in the `test` script is because `minimist` didn't have a nice way to get all arguments in a list when you wanted to forward them, so I ended up just explicitly passing through `--watch`.

Reply:
Good point. It had some weird way of setting it and forwarding it before. I think that was probably a limitation of `minimist` and the way it was setup.